app/views/flood_risk_engine/enrollments/steps/_local_authority_address.html.erb

```diff
@@ -11,25 +11,35 @@
   <%= t('.select_from_list_label') %>
 </label>
 
-<%# Create prompt manually cos this too flakey to use - e.g disappears on Back
-{ prompt: t('.num_addresses_found', num:  form.address_list.count) } %>
-
-<% form.address_list.unshift({ moniker: t('.num_addresses_found', num:  form.address_list.count) }) %>
-
 <fieldset>
   <%= form_group_and_validation(form, :uprn) do %>
-    <%= f.select :uprn,  options_for_select( form.address_list.collect { |a| [ a[:moniker], a[:uprn] ] }), {},
-                 { include_blank: false,
-                   class: 'form-control form-control-select',
-                   id: 'address_match_selection'
-                 }
+    <%= f.select(
+          :uprn,
+          options_for_select( form.address_list.collect { |a| [ a[:moniker], a[:uprn] ] }),
+          {
+            include_blank: t('.num_addresses_found', num:  form.address_list.count)
```

Comment:
Two points - you used `prompt` (on my suggestion) and the behaviour of `prompt` is precisely as you describe - it only appears if the underlying attribute is empty. It will disappear if you access the form after a value has been assigned to the attribute. If the text needs to always be there, then you need to use `include_blank` and not `prompt` (my bad suggestion to use `prompt` without also suggesting `include_blank`).

Also the include_blank needs to be in the the right hash. I did test the different scenarios.